crates/sui-types/src/crypto.rs

```diff
@@ -478,6 +480,88 @@ where
     (kp.public().into(), kp)
 }
 
+/// Ed25519 follows SLIP-0010 using hardened path: m/44'/784'/0'/0'/{index}'
+/// Secp256k1 follows BIP-32 using path where the first 3 levels are hardened: m/54'/784'/0'/0/{index}
+/// Note that the purpose for Secp256k1 is registered as 54, to differentiate from Ed25519 with purpose 44.
+pub fn derive_key_pair_from_path(
+    seed: &[u8],
```

Comment:
In a future PR let's have the `seed` as a distinct struct vs plain [u8]. Fine to merge as it is now; this will be a generic crypto refactoring.